src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/utbetalingsoppdrag/ØkonomiUtils.kt

```diff
@@ -103,6 +103,38 @@ object ØkonomiUtils {
         }
     }
 
+    fun utbetalingsperiodeForOpphør(forrigeTilkjentYtelse: TilkjentYtelse?,
+                                    nyTilkjentYtelseMedMetaData: TilkjentYtelseMedMetaData): Utbetalingsperiode? {
+        val nyTilkjentYtelse = nyTilkjentYtelseMedMetaData.tilkjentYtelse
+        validerStartdato(forrigeTilkjentYtelse, nyTilkjentYtelse)
+
+        // hvis det ikke finnes tidligere andel så kan vi ikke opphøre noe
+        val sisteForrigeAndel = forrigeTilkjentYtelse?.sisteAndelIKjede ?: return null
+        val forrigeAndeler = andelerUtenNullVerdier(forrigeTilkjentYtelse)
+        val forrigeMaksDato = forrigeAndeler.map { it.tilOgMed }.maxOrNull()
+        val nyeAndeler = andelerUtenNullVerdier(nyTilkjentYtelse)
+
+        // TODO denne kan fjernes når startdato blir not null
+        if (forrigeTilkjentYtelse.startdato == null || nyTilkjentYtelse.startdato == null) {
+            error("Må ha med startdato på ny og gammel tilkjent ytelse")
+        }
+        if (nyTilkjentYtelse.startdato < forrigeTilkjentYtelse.startdato) {
```

Comment:
Tror vi kan navngi denne for å forstå "hvorfor" vi returnerer her:
F.eks.
`if(nyeAndelerErstatterAlleTidligereAndeler(nytilkjentYtelse, forrigeTilkjentYtelse))` ?

Reply:
Tja, det kan skje ellers også, trekker det ut i egen variabel `skalOpphøreFørTidligereStartdato`